### ISSUE TITLE
Export to CSV while in server mode

### ISSUE THREAD

**Issue by mitar** [nicolargo/glances#1527]
#### Description

It looks like it is not possible to export data when running in server mode? That is surprising because when I read it in documentation I though that is the main use case. That you can run a server daemon and configure it so that it exports everything recorded to a file for preservation. And then you can connect with a client to see current state.

It seems you have to connect with a client to export recorded data, but why couldn't just server do that as well? So then there would be no need for me to periodically connect and export.

#### Versions

* Glances & psutil (glances -V): Glances v3.1.1 with psutil v5.6.3
* Operating System (lsb_release -a): Ubuntu 18.04.2 LTS

**Comment by nicolargo** [comment]
For the moment, export is only available in standalone or client mode...

**Comment by mitar** [comment]
Where does server mode save recorded data? In memory?

**Comment by nicolargo** [comment]
Yes in memory. You can configure the buffer size here: https://github.com/nicolargo/glances/blob/develop/conf/glances.conf#L8-L10

**Comment by mitar** [comment]
Yes, I have seen that you can configure it.

I really thought that all integration with CSV and InfluxDB and stuff like that can work on the server, that you can export all this data in real-time into those and then use them.

**Comment by nicolargo** [comment]
By design, Glances server has a very low CPU footprint because it grab data only when a client ask for it (not every 3 seconds). If we allows user to serve and export i have to change this behavior on server side.

But i hear your need. I am going to think to a new mode to mix server and export mode...

Any ideas are welcomes.

**Comment by MetallicAchu** [comment]
Is it possible to create 2 systemd services, for example glances.webserver and glances.exporter and use each of them according to the respective examples on https://github.com/nicolargo/glances/wiki/Start-Glances-through-Systemd ?

Are there any limitations to potentially running two instances while one is the webserver?

Thanks!!

**Comment by nicolargo** [comment]
@MetallicAchu Yes it is possible to run two Glances instances.

For the CPU consumption, it is strange, on my system i am around 7% not 20%.... Try to disable the sensors plugin.

**Comment by MetallicAchu** [comment]
I made 2 services, one for web server and one for export to prometheus every 10 seconds
I tested it with ps aux | grep glances --- because ps aux gives the average CPU consumption over the running time.

Tested for almost 20 minutes:
 16060 10.2  1.5 1373528 62524 ?       Ssl  12:10  18:57 /usr/bin/python3 /usr/bin/glances -t 10 --quiet --export prometheus

Where 10.2 is CPU % over time run.

I'll let it run for a while more and update again :)

**Comment by MetallicAchu** [comment]
I wanted to update that with glances v3.2.0 I'm experiencing 2.5% CPU on average, so the change is most welcomed!
Thank you!

**Comment by crile** [comment]
Same use case here, I start the server and the client for the export. I think it do the job
```
glances --config glances.conf --server
glances --config glances.conf --client localhost --export csv --export-csv-file export.csv
```
However I noticed that with the option `--quiet`, glances use 100% of CPU load.
The client use 70% and server instance use the rest.

So I replace the client command by :

`glances --config glances.conf --client localhost --export csv --export-csv-file export.csv > /dev/null`

I can't try with client instance on another machine.

Glances v3.2.3.1 with PsUtil v5.8.0

**Comment by joshua-lehmann** [comment]
Hi @nicolargo First of all thanks for your great tool! You marked this issue as solved with Release 4.0.0, but I was not able to run the server and export simultaneously. My goal is to monitor my raspberry PI with the webserver mode for the current/live stats and to also export the stats and store them for 7 days in Influx DB using the export mechanism of glances, so I can see peaks etc within a week. However, I was not able to get both the export and server mode working with docker-compose. Is this supported since v4? And if not, what would be you recommend way to get the export influx dB export and occasionally live check working at the same time?

**Comment by c0delama** [comment]
Coming from "Export is only available in standalone or client mode” and also stumbled over this issue. Continuously pushing data to InfluxDB2 so i could retrospectively check it out would be my main use case. Nevertheless keep up the great work so far 👍

**Comment by YaibaToKen** [comment]
Out of curiosity, has there been any news on this? 
Is it possible to raise a container with glances running just the glances browser and raise another container in client mode, connected to the browser one?

**Comment by nicolargo** [comment]
Done on the develop-v5 branch.

Will be release in Glances 5.0.0.
